UPGRADE.md

```diff
@@ -30,13 +30,73 @@ mysqldump -u YOUR_ILIOS_DATABASE_USERNAME -h YOUR_ILIOS_HOSTNAME -R -p YOUR_ILIO
+3. Now that you're sure you're ready to migrate, navigate into your Ilios application's root directory (eg, /web/ilios3/ilios) and examine the file at app/config/parameters.yml.  Look for the value that corresponds to file_system_storage_path.  This is where your learning materials will be moved.
+4. Now, check to see where your *CURRENT* Ilios 2 learning materials reside.  If your Ilios 2 installation is at /web/ilios2/htdocs, your current learning material are probably in '/web/ilios2/htdocs/learning_materials'. If this is the case, your learning materials reside in '/web/ilios2/htdocs' -- this is the 'Path to Ilios 2' location that you should enter in the following migration command:
+
+Now, while still in your Ilios application's root directory, run this command:
+
+```bash
+sudo -u apache bin/console ilios:setup:migrate-learning-materials /web/ilios2/htdocs --env=prod
+```
+
+Make sure that you have entered YOUR 'Path to Ilios 2' location and also do not forget to add '--env=prod' to end of the command!  When you hit enter, you learning materials will be copied to the location as listed in your parameters.xml file.
+
+This process can take a while to complete, depending on the number/size of the Learning Materials being copied.
+
+If you would like to verify that your new learning materials are correctly stored, you can run the following command when the migration is complete:
+
+```bash
+sudo -u apache bin/console ilios:setup:validate-learning-materials /web/ilios2/htdocs --env=prod
+```
+
+## Cleaning up the text of your Ilios 2.x Topics
+
+One of the biggest complaints of users of Ilios 2.x, regarding the application, was the poor-formatting of the text in item titles and descriptions. While Ilios 2.x did provide a WYSIWYG editor to its users, too many of the users chose to paste text directly into the Ilios text fields from whatever application they were already in (usually MS Word or some other MS Office application), instead of re-entering the text using the in-app editor.  This caused a real problem with the formatting of text in Ilios, so we've updated the handling of text in Ilios 3 and limited the formatting options to only those most-used in Ilios 2.  Doing so, while making new and future text entries format quite nicely, has made the old, over-formatted, pasted-from-Word text format terribly, rendering much of it unreadable.
+
+We have provided a solution to this problem with a fix that will allow you to quickly remove all but only the allowed characters from your item title and description text.  To apply this fix and reformat all your text to only use the proper formatting native to Ilios 3, do the following:
+
+* Navigate to your Ilios application root directory ('/web/ilios3/ilios' for example) and run the following command:
+
+```bash
+sudo -u apache bin/console ilios:maintenance:cleanup-strings --env=prod
```

Comment:
Several commands here which all have to be run:

`sudo -u apache bin/console ilios:maintenance:cleanup-strings --env=prod --objective-title`
`sudo -u apache bin/console ilios:maintenance:cleanup-strings --env=prod --learningmaterial-description`
`sudo -u apache bin/console ilios:maintenance:cleanup-strings --env=prod --learningmaterial-note `
`sudo -u apache bin/console ilios:maintenance:cleanup-strings --env=prod --session-description`